packages/block-editor/src/components/block-list/test/block-list-context.native.js

```diff
@@ -0,0 +1,266 @@
+		const defaultBlockLayouts = cloneDeep( BLOCKS_LAYOUTS_DATA );
+		const currentBlockLayouts = deleteBlockLayoutByClientId(
+			defaultBlockLayouts,
+			DEEP_NESTED_ID
+		);
+
+		const findDeletedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts,
+			DEEP_NESTED_ID
+		);
+
+		expect( findDeletedBlock ).toEqual(
+			expect.not.objectContaining( {
+				clientId: DEEP_NESTED_ID,
+			} )
+		);
+	} );
+} );
+
+describe( 'updateBlocksLayouts', () => {
+	it( "adds a new block's layout data at root level im an empty object", () => {
+		const {
+			blocksLayouts,
+			findBlockLayoutByClientId,
+			updateBlocksLayouts,
+		} = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = cloneDeep( blocksLayouts );
+		const BLOCK_CLIENT_ID = PARAGRAPH_BLOCK_LAYOUT_DATA.clientId;
+
+		updateBlocksLayouts( currentBlockLayouts, PARAGRAPH_BLOCK_LAYOUT_DATA );
+
+		const findAddedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			BLOCK_CLIENT_ID
+		);
+
+		expect( findAddedBlock ).toEqual(
+			expect.objectContaining( { clientId: BLOCK_CLIENT_ID } )
+		);
+	} );
+
+	it( "adds a new block's layout data at root level with inner blocks", () => {
+		const {
+			findBlockLayoutByClientId,
+			updateBlocksLayouts,
+		} = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = {
+			current: cloneDeep( BLOCKS_LAYOUTS_DATA ),
+		};
+		const PARENT_BLOCK_CLIENT_ID = GROUP_BLOCK_LAYOUT_DATA.clientId;
+
+		// Add parent block
+		updateBlocksLayouts( currentBlockLayouts, GROUP_BLOCK_LAYOUT_DATA );
+
+		const findAddedParentBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			PARENT_BLOCK_CLIENT_ID
+		);
+
+		expect( findAddedParentBlock ).toEqual(
+			expect.objectContaining( { clientId: PARENT_BLOCK_CLIENT_ID } )
+		);
+
+		// Add inner block to it's parent
+		updateBlocksLayouts( currentBlockLayouts, {
+			...PARAGRAPH_BLOCK_LAYOUT_DATA,
+			rootClientId: PARENT_BLOCK_CLIENT_ID,
+		} );
+
+		const findAddedInnerBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			PARAGRAPH_BLOCK_LAYOUT_DATA.clientId
+		);
+
+		expect( findAddedInnerBlock ).toEqual(
+			expect.objectContaining( {
+				clientId: PARAGRAPH_BLOCK_LAYOUT_DATA.clientId,
+			} )
+		);
+	} );
+
+	it( "adds a new block's layout data at a deep level", () => {
+		const {
+			findBlockLayoutByClientId,
+			updateBlocksLayouts,
+		} = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = {
+			current: cloneDeep( BLOCKS_LAYOUTS_DATA ),
+		};
+
+		// Add block layout data to it's parents inner blocks
+		updateBlocksLayouts( currentBlockLayouts, {
+			...PARAGRAPH_BLOCK_LAYOUT_DATA,
+			rootClientId: NESTED_WITH_INNER_BLOCKS_ID,
+		} );
+
+		const findAddedInnerBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			PARAGRAPH_BLOCK_LAYOUT_DATA.clientId
+		);
+
+		expect( findAddedInnerBlock ).toEqual(
+			expect.objectContaining( {
+				clientId: PARAGRAPH_BLOCK_LAYOUT_DATA.clientId,
+			} )
+		);
```

Comment:
Similar to previous comments, we could also check the `rootClientId` here:
```suggestion
		expect( findAddedInnerBlock ).toEqual(
			expect.objectContaining( {
				clientId: PARAGRAPH_BLOCK_LAYOUT_DATA.clientId,
				rootClientId: NESTED_WITH_INNER_BLOCKS_ID,
			} )
		);
```